### PR TITLE
Add PodWrapper functions for scheduler testing

### PR DESCRIPTION
**Pull request by yuanchen8911**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

1. Add additional `PodWrapper` functions to facilitate scheduler testing. 
    1.1 `Annotation(key, value string)` adds a pair of (key, value) to pod.Annotations.
    1.2 `CreationTimestamp(t metav1.Time)` sets a pod's creationTimestamp.
    1.3 `Condition(type v1.PodConditionType, status v1.ConditionStatus, reason string)` adds  a condition to pod.Status.Conditions.
2. Replace ad-hoc pod building with `PodWrapper.MakePod` in `pkg/scheduler/internal/queue/scheduler_queue_test.go`.

#### Which issue(s) this PR fixes:

None

#### Special notes for your reviewer:

We should use `PodWrapper.MakePod` in other tests. 

#### Does this PR introduce a user-facing change?

```release-note
NONE

```
#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by yuanchen8911** [comment]
sig/scheduling

File: pkg/scheduler/testing/wrappers.go [hunk 1]
```diff
@@ -266,6 +281,12 @@ func (p *PodWrapper) NominatedNodeName(n string) *PodWrapper {
 	return p
 }
 
+// Condition adds a `conditon(Type, Status, Reason)` to .status.Conditions.
```
Comment (denkensk) [inline pkg/scheduler/testing/wrappers.go:284]:
```suggestion
// Condition adds a `conditon(Type, Status, Reason)` to .Status.Conditions.
```

**Review by denkensk — COMMENTED** [review]
Thanks. 
Could you check the other files in `pkg/scheduler` and replace `&v1.Pod{}` with `makePod` if exists ?

**Comment by yuanchen8911** [comment]
> Could you check the other files in pkg/scheduler and replace &v1.Pod{} with makePod if exists ?


A lot😄. Replacing all will require some efforts. Can it be a separate PR?

File: pkg/scheduler/testing/wrappers.go [hunk 2]
```diff
@@ -266,6 +281,12 @@ func (p *PodWrapper) NominatedNodeName(n string) *PodWrapper {
 	return p
 }
 
+// Condition adds a `conditon(Type, Status, Reason)` to .Status.Conditions.
```
Comment (kerthcet) [inline pkg/scheduler/testing/wrappers.go:284]: s/conditon/condition/

Comment (yuanchen8911) [reply pkg/scheduler/testing/wrappers.go:284]: Thank you so much, @kerthcet!

**Comment by Huang-Wei** [comment]
/lgtm
/approve

Thanks @yuanchen8911 .

**Pull request merged**